src/lib.rs

```diff
@@ -160,7 +160,7 @@ use num_traits::{One, Zero};
 use std::borrow::Borrow;
 use std::collections::HashMap;
 use std::hash::Hash;
-use std::iter;
+use std::iter::{self, Sum};
```

Comment:
```suggestion
use std::iter;
```

Given the above we can revert this change.

Reply:
Ok, sounds good.  I pushed another commit reverting that part of the original.